addons/web/static/src/model/model.js

```diff
@@ -154,23 +165,40 @@ export function useModelWithSampleData(ModelClass, params, options = {}) {
             model.orm = sampleORM;
             await model.load(searchParams);
             model.orm = orm;
+            model.useSampleModel = true;
         } else {
             useSampleModel = false;
             model.useSampleModel = useSampleModel;
         }
-        if (started) {
+        model.whenReady.resolve(); // resolve after the first successful load
+        if (status(component) === "mounted") {
             model.notify();
```

Comment:
done in whenReady.then already

Reply:
only the first time, but we need to call notify all the time (except if we're not mounted yet)